rest/src/main/java/org/jboss/pnc/rest/provider/BuildRecordProvider.java

```diff
@@ -660,63 +696,76 @@ public BuildRecordRest getLatestBuildRecord(Integer configId, boolean executedOn
+            if (!buildConfigurationAuditedIdRevs.isEmpty()) {
+                dbAndPredicatesList.add(BuildRecordPredicates.withBuildConfigurationIdRev(buildConfigurationAuditedIdRevs));
+            } else {
+                dbAndPredicatesList.add(Predicate.nonMatching());
             }
-        }
 
-        if (!StringUtils.isEmpty(andFindByBuildConfigurationName)) {
-            //add steam condition
+        } else if (!StringUtils.isEmpty(orFindByBuildConfigurationName)) {
+            // add steam condition
+            StringBuilder sb = new StringBuilder(",(buildConfigurationName=like=").append(orFindByBuildConfigurationName).append(")");
+
             if (StringUtils.isEmpty(combinedQueries)) {
-                combinedQueries = "(buildConfigurationName==" + andFindByBuildConfigurationName + ")";
+                combinedQueries = sb.deleteCharAt(0).toString();
             } else {
-                combinedQueries = "(" + combinedQueries + ");(buildConfigurationName==" + andFindByBuildConfigurationName + ")";
+                combinedQueries = sb.insert(0, ")").insert(0, combinedQueries).insert(0, "(").toString();
             }
 
-            //add DB predicate
-            List<BuildConfigurationAudited> buildConfigurationAuditeds = buildConfigurationAuditedRepository
-                    .searchForBuildConfigurationName(andFindByBuildConfigurationName);
-            if (!buildConfigurationAuditeds.isEmpty()) {
-                dbAndPredicatesList.add(
-                        BuildRecordPredicates.withBuildConfigurationIdRev(
-                                buildConfigurationAuditeds.stream().map(bca -> bca.getIdRev()).collect(Collectors.toList())
-                        )
-                );
-            } else {
-                dbAndPredicatesList.add(Predicate.nonMatching());
+            // add DB predicate
+            List<IdRev> buildConfigurationAuditedIdRevs = buildConfigurationAuditedRepository
+                    .searchIdRevForBuildConfigurationName(orFindByBuildConfigurationName);
+            if (!buildConfigurationAuditedIdRevs.isEmpty()) {
+                dbOrPredicateList.add(BuildRecordPredicates.withBuildConfigurationIdRev(buildConfigurationAuditedIdRevs));
             }
         }
 
-        Set<BuildRecordRest> running = nullableStreamOf(buildCoordinator.getSubmittedBuildTasks())
-                .map(this::createNewBuildRecordRest)
-                .filter(rsqlPredicateProducer.getStreamPredicate(BuildRecordRest.class, combinedQueries))
-                .sorted(sortInfoProducer.getSortInfo(sort).getComparator())
-                .collect(Collectors.toSet());
+        List<BuildTask> buildTasks = buildCoordinator.getSubmittedBuildTasks();
+        Set<IdRev> buildConfigAuditedIdRevsOfBuildTasks = nullableStreamOf(buildTasks).map(buildTask -> buildTask.getBuildConfigurationAudited().getIdRev()).collect(Collectors.toSet());
```

Comment:
The old code invoked method createNewBuildRecordRest, which invoked **buildConfigurationAuditedRepository.queryById(IdRev)**. That method queried 1 by 1 all the BuildConfigurationAudited associated with eventually running BuildTasks; it queried the BuildRecords associated with each retrieved BuildConfiguration; it fetched the GenericParameters of each BuildConfiguration retrieved. Moreover, the BuildRecords queried, while used in inner methods to construct the BuildConfigurationAudited objects, were never used.
Considering **20** running builds, the number of queries done were:
20 x (1 +1 +1) = **60** <-- 20 running builds x (1 query for BuildConfigurationAudited; 1 query for all the BuildRecords; 1 query to fetch the GenericParameters)

The new code retrieves all the IdRevs of the BuildConfigurationAudited associated with the eventually running BuildTasks and retrieves with 1 query all the BuildConfigurationAudited objects; does not query the BuildRecords (not used anyway); queries the GenericParameters.
Thus, again considering **20** running builds, the number of queries done are: 
1 + 0 + 20 x 1 = **21** <-- 1 query for all 20 BuildConfigurationAudited; no query for BuildRecords; 1 query for each rBuildConfigurationAudited to fetch GenericParameters.